lib/shopify_api/auth/oauth.rb

```diff
@@ -47,7 +47,7 @@ def begin_auth(shop:, redirect_path:, is_online: true, scope_override: nil)
 
           query_string = URI.encode_www_form(query)
 
-          auth_route = "https://#{shop}/admin/oauth/authorize?#{query_string}"
+          auth_route = "https://#{shop.sub(".my.shop.dev", ".dev-api.shop.dev")}/admin/oauth/authorize?#{query_string}"
```

Comment:
I'll still work towards making this unnecessary.

Reply:
Sorbet makes it clumsier than it should be, but here's an alternative that uses standard endpoints:

```rb
          auth_route = if defined?(DevServer)
            admin_web = T.unsafe(Object.const_get("DevServer")).new("web") # rubocop:disable Sorbet/ConstantsFromStrings
            admin_host = admin_web.unsafe_unchecked_host(nonstandard_host_prefix: "admin")

            "https://#{admin_host}/store/#{shop.split(".").first}/oauth/authorize?#{query_string}"
          else
            "https://#{shop}/admin/oauth/authorize?#{query_string}"
          end
```

Reply:
@GoodForOneFare Re-tested the installation flow with this new solution, it works perfectly.
Pushed a commit to use this solution and moved it to a dedicated method with a comment for clarity.